### PR TITLE
Rename IModelFinalizedConvention to IModelFinalizingConvention

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Introduce IModelFinalizedConvention that don't change the model

Part of #12846

### CODE REVIEW

File: src/EFCore/Metadata/Internal/Model.cs [hunk 1]
```diff
@@ -882,8 +890,13 @@ public virtual void SetPropertyAccessMode(PropertyAccessMode? propertyAccessMode
         /// </summary>
         public virtual IModel FinalizeModel()
         {
-            var finalModel = (Model)ConventionDispatcher.OnModelFinalized(Builder)?.Metadata;
-            return finalModel?.MakeReadonly() ?? this;
+            IModel validatedModel = ConventionDispatcher.OnModelFinalized(Builder)?.Metadata;
+            if (validatedModel != null)
+            {
+                validatedModel = ConventionDispatcher.OnModelFinalized(validatedModel);
```
Comment (smitpatel) [inline src/EFCore/Metadata/Internal/Model.cs]: Why does this need to run again?

**Comment by ajcvickers** [comment]
@AndriySvyryd Motivation for changing the name here? Is it purely to make the name more "correct" or do we functionally need to change?

**Comment by AndriySvyryd** [comment]
@ajcvickers We need to have conventions that run after the OnModelFinalized conventions for two reasons:

1. Moving `ValidatingConvention` there will make it easier for providers and users to add conventions as they need to run before `ValidatingConvention`
2. We need a hook for model transformations that don't change the configuration, such as adding a relational model or switching to a runtime-optimized implementation.

File: src/EFCore/Metadata/Internal/Model.cs (hunk 1 above)
Comment (AndriySvyryd) [reply src/EFCore/Metadata/Internal/Model.cs]: Typo, thanks

**Pull request merged**